features/dd-sdk-android-trace/src/main/java/com/datadog/trace/common/writer/ListWriter.java

```diff
@@ -37,6 +37,9 @@ public boolean accept(List<DDSpan> trace) {
   public List<DDSpan> firstTrace() {
     return get(0);
   }
+  public List<DDSpan> secondTrace() {
```

Comment:
yup, they are using groovy in their tests and they are able to access these private properties directly.

Reply:
let's maybe do the same? write reflection wrappers to access all this. Because otherwise we make quite a lot of internal things public just for the tests, not sure if it opens the door for the misuse on the client side.

Reply:
it shouldn't honestly, there are plenty of places like these in their api's.

Reply:
I think it makes sense if you think about this code as an agent - it is attached to the process and is not supposed to be used as a SDK.

But in our case it becomes a SDK, and we should care much more about things which should be open vs closed.

Reply:
actually this class is only used for testing purposes on their end, it blocks the thread so to make sure the traces are consumed. I think this one is safe.